eclair-core/src/main/scala/fr/acinq/eclair/wire/protocol/MessageOnion.scala

```diff
@@ -127,16 +128,16 @@ object MessageOnionCodecs {
 
   val blindedRouteCodec: Codec[BlindedRoute] = (("firstNodeId" | publicKey) :: ("blinding" | publicKey) :: ("path" | list(replyHopCodec).xmap[Seq[BlindedNode]](_.toSeq, _.toList))).as[BlindedRoute]
 
-  private val replyPathCodec: Codec[ReplyPath] = variableSizeBytesLong(varintoverflow, blindedRouteCodec).as[ReplyPath]
+  private val replyPathCodec: Codec[ReplyPath] = tlvField(blindedRouteCodec.as[ReplyPath])
 
-  private val encryptedDataCodec: Codec[EncryptedData] = variableSizeBytesLong(varintoverflow, bytes).as[EncryptedData]
+  private val encryptedDataCodec: Codec[EncryptedData] = tlvField(bytes.as[EncryptedData])
 
   private val onionTlvCodec = discriminated[OnionMessagePayloadTlv].by(varint)
     .typecase(UInt64(2), replyPathCodec)
     .typecase(UInt64(4), encryptedDataCodec)
-    .typecase(UInt64(64), variableSizeBytesLong(varintoverflow, OfferCodecs.invoiceRequestTlvCodec.as[InvoiceRequest]))
-    .typecase(UInt64(66), variableSizeBytesLong(varintoverflow, OfferCodecs.invoiceTlvCodec.as[Invoice]))
-    .typecase(UInt64(68), variableSizeBytesLong(varintoverflow, OfferCodecs.invoiceErrorTlvCodec.as[InvoiceError]))
+    .typecase(UInt64(64), tlvField(OfferCodecs.invoiceRequestTlvCodec.as[InvoiceRequest]))
+    .typecase(UInt64(66), tlvField(OfferCodecs.invoiceTlvCodec.as[Invoice]))
+    .typecase(UInt64(68), tlvField(OfferCodecs.invoiceErrorTlvCodec.as[InvoiceError]))
```

Comment:
I agree, it's more a refactoring than a way to make it less footgunny. Can't think of a better way either though.

Reply:
Yeah I couldn't find anything better either...but since two different people made a mistake recently (forgetting the `variableSizeByteLong` part), I figured we had to make it more explicit, and I believe this is more explicit: you see that all tlv codecs are wrapped inside a `tlvField` so you understand that if you're adding a tlv, you should do the same. When previously you just saw that other codecs (and not all of them) were using `variableSizeBytesLong`, it didn't jump at you that this was required for all tlvs, did it?

Also I harmonized tlvs by removing the ones that were using `variableSizeBytesLong` in a less obvious ways (with fixed length, or with the length-prefixed truncated int codecs).